cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/VerificationStage.java

```diff
@@ -169,6 +170,19 @@ public void process(ApplicationDeployable input) throws Exception {
       }
     }
 
+    // TODO StreamSizeSchedules should be resilient to stream inexistence [CDAP-1446]
+    for (Map.Entry<String, ScheduleSpecification> entry : specification.getSchedules().entrySet()) {
+      Schedule schedule = entry.getValue().getSchedule();
+      if (schedule instanceof StreamSizeSchedule) {
+        StreamSizeSchedule streamSizeSchedule = (StreamSizeSchedule) schedule;
+        String streamName = streamSizeSchedule.getStreamName();
+        if (!specification.getStreams().containsKey(streamName)) {
```

Comment:
Ok, I thought we said this morning that we could go with the assumption that the stream had to be in the app specification for now, and that we would allow the stream to not exist, etc, later.
But I can make the extra check to the store to make sure that the stream exists.